src/wallet/coincontrol.h

```diff
@@ -76,8 +77,24 @@ class CCoinControl
         vOutpoints.assign(setSelected.begin(), setSelected.end());
     }
 
+    void AddKnownCoins(const CInputCoin& coin)
+    {
+        knownCoins.insert(std::make_pair(coin.outpoint, coin));
+    }
+
+    boost::optional<CInputCoin> FindKnownCoin(const COutPoint& outpoint) const
```

Comment:
Can we avoid `boost::optional`?
Maybe `std::shared_ptr<CInputCoin>` should do it?
Maybe switch the map keys to std::pair<uint256, int>?